### PR TITLE
FEATURE: Add option to delete all replies of flagged post

### PR DESCRIPTION
**Pull request by gschlager**
Shows a dialog when the flagged posts gets deleted, it isn't the first post in a topic and it has replies.

image

### CODE REVIEW

File: app/assets/javascripts/admin/models/flagged-post.js.es6 [hunk 1]
```diff
@@ -62,11 +62,74 @@ export default Post.extend({
...
+      return action().catch(popupAjaxError);
+    }
+  },
+
+  _hasDeletableReplies() {
+    return this.get('post_number') > 1 && this.get('reply_count') > 0;
+  },
+
+  _actOnFlagAndDeleteReplies(action) {
+    return new Ember.RSVP.Promise((resolve, reject) => {
```
Comment (tgxworld) [inline app/assets/javascripts/admin/models/flagged-post.js.es6]: Not sure if I'm missing something but why do we need to return a promise here? We don't seem to be using the promise anywhere.

File: app/assets/javascripts/admin/models/flagged-post.js.es6 [hunk 2]
```diff
@@ -62,11 +62,74 @@ export default Post.extend({
...
+      return ajax(`/posts/${this.id}/reply-ids/all.json`).then(replies => {
```
Comment (tgxworld) [inline app/assets/javascripts/admin/models/flagged-post.js.es6]: I think we should avoid this extra request and do it all on the server side. We can use`reply_count` to replace `replies.length`.

File: app/assets/javascripts/admin/models/flagged-post.js.es6 [hunk 3]
```diff
@@ -62,11 +62,74 @@ export default Post.extend({
...
 
   agreeFlags(actionOnPost) {
-    return ajax('/admin/flags/agree/' + this.id, { type: 'POST', cache: false, data: { action_on_post: actionOnPost } }).catch(popupAjaxError);
+    const action = () => {
+      return ajax('/admin/flags/agree/' + this.id, {
+        type: 'POST', cache: false, data: { action_on_post: actionOnPost }
+      });
+    };
+
+    if (actionOnPost === 'delete' && this._hasDeletableReplies()) {
```
Comment (tgxworld) [inline app/assets/javascripts/admin/models/flagged-post.js.es6]: I think this should really just be a param that we pass to `/admin/flags/agree` to make it delete all the replies as well.

File: app/assets/javascripts/admin/models/flagged-post.js.es6 (hunk 1 above)
Comment (gschlager) [reply app/assets/javascripts/admin/models/flagged-post.js.es6]: It's waiting on the promise in the `removeAfter` calls: https://github.com/discourse/discourse/blob/ce97759418255af409693b95ca23f1274df8cd51/app/assets/javascripts/select-kit/components/admin-delete-flag-dropdown.js.es6#L68-L76

File: app/assets/javascripts/admin/models/flagged-post.js.es6 (hunk 2 above)
Comment (gschlager) [reply app/assets/javascripts/admin/models/flagged-post.js.es6]: `reply_count` is only the number of direct replies and `replies.length` is the number of all replies, so replacing won't work. I guess I could replace the second call by sending the reply ids to the server.

**Pull request merged**